### DISCUSSION TITLE
"Convert to interpolated string" for string + any object as warning

### DISCUSSION THREAD

**Discussion by ufcpp** [dotnet/roslyn#70957] [Ideas] [answered]
Today, the following code was discussed on X (Twitter) in Japan.

```cs
Console.WriteLine(1 + "2"); // 12
```

This code, familiar to senior C# developers, seems strange to beginners. They say, "It doesn't look like a strongly typed language; it looks like JavaScript." Also, the following code looks confusing even for senior devs.

```cs
Console.WriteLine(1 + 1 + "1"); // 21
Console.WriteLine("1" + 1 + 1); // 111
```

Even today, code fixes are available for this code.

![image](https://github.com/dotnet/roslyn/assets/3760955/29142b7d-360c-4d87-86ae-b395cc1c7440)

What I would like to suggest here is to add an analyzer to this, so that the warning level can be changed in .editorconfig. I want to treat this as a warning.

**Comment by CyrusNajmabadi** [comment]
Can you move this to roslyn-analyzers?  I can't seem to move it myself.  Thanks!

**Reply by ufcpp** [reply]
I couldn't do it too. I will re-create the issue there.

**Reply by ufcpp** [reply, marked as answer]
Done https://github.com/dotnet/roslyn-analyzers/issues/7057
